examples/Example.java

```diff
@@ -18,19 +18,15 @@ public static void main(String[] args) throws IOException {
 
     Request request = new Request();
     request.baseUri = "api.sendgrid.com";
-    Map<String,String> requestHeaders = new HashMap<String, String>();
-    requestHeaders.put("Authorization", "Bearer " + System.getenv("SENDGRID_API_KEY"));
-    request.headers = requestHeaders;
+    request.getHeaders().put("Authorization", "Bearer " + System.getenv("SENDGRID_API_KEY"));
 
     Response response = new Response();
 
     // GET Collection
     request.method = Method.GET;
     request.endpoint = "/v3/api_keys";
-    Map<String,String> queryParams = new HashMap<String, String>();
-    queryParams.put("limit", "100");
-    queryParams.put("offset", "0");
-    request.queryParams = queryParams;
+    request.getQueryParams().put("limit", "100");
+    request.getQueryParams().put("offset", "0");
```

Comment:
Can we use a setter here instead?

e.g. `request.addQueryParam("offset", "0");`